Implemented Page Object Model

aneghina opened the pull request.

dzarubingrid commented on pom/login.spec.ts:
```diff
@@ -0,0 +1,25 @@
+    async close() {
+        await this.page.close();
+    }
+
```
Unnecessary empty line

dzarubingrid commented on pom/login.spec.ts:
```diff
@@ -0,0 +1,25 @@
+import { type Locator, type Page } from "@playwright/test"
+
+export class LoginPage {
```
The class is called LoginPage, so it must be placed in a file called loginPage.ts

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,19 @@
-  });
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  await loginPageTest.emailField.fill('ag.neghina@gmail.com');
```
Let's extract lines 7-9 in a separate function in the LoginPage class. The usage would be await loginPage.login('ag.neghina@gmail.com', 'An2024grid!');

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,19 @@
+  await loginPageTest.emailField.fill('ag.neghina@gmail.com');
+  await loginPageTest.passwordField.fill('An2024grid!');
+  await loginPageTest.loginButton.click();
+  await expect(loginPageTest.loginButton).toBeVisible();
```
No bueno. We must validate for sure that we are logged in

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,19 @@
+  await loginPageTest.passwordField.fill('An2024grid!');
+  await loginPageTest.loginButton.click();
+  await expect(loginPageTest.loginButton).toBeVisible();
+  await loginPageTest.close();
```
There is no need to close the page after the test - Playwright will do it

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,17 @@
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  await loginPageTest.login('ag.neghina@gmail.com', 'An2024grid!');
+  await loginPageTest.loginButton.click();
```
Please move this line into loginPage.login(). As the function name implies, user must be logged in when the function is called

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,17 @@
-    await expect(page).toHaveURL(/.*failure/);
-    await page.close();
-  });
+  const loginPageTest = new LoginPage(page);
```
loginPageTest => loginPage?

dzarubingrid commented on tests/test.spec.ts:
```diff
@@ -1,15 +1,17 @@
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  expect(await page.screenshot()).toMatchSnapshot();
+  await loginPageTest.close();
```
There is no need to close pages :)

dzarubingrid commented on pom/loginPage.ts:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    async goto() {
+        await this.page.goto('https://www.facebook.com');
```
Please move the url into the playwright config (baseUrl)

dzarubingrid commented on pom/loginPage.ts:
```diff
@@ -0,0 +1,31 @@
+import { type Locator, type Page } from "@playwright/test"
+
+export class LoginPage {
+
```
Unnecessary blank line

aneghina commented:
In this new commit, I have:
- Deleted empty lines and unnecessary lines of code
- Moved the URL into the Playwright config
- Changed the name of the constant to 'loginPage'
- Added a 'click()' method to the 'login' function
